### PR TITLE
docs: stop tracking placeholder documentation

### LINKED ISSUE #26205: Stop checking in generated docs
Instead, have a presubmit test that verifies that the docs can be generated. This will reduce the number of conflicts & rebases, therefore increase speed through the queue.

Confirmed with @bgrant0607 that nothing depends on these docs at the moment.

@johndmulhausen fyi, some manual process might have to change once this is in.

### PR DESCRIPTION
**Pull request by neolit123**
**What type of PR is this?**
/kind cleanup
/kind documentation

**What this PR does / why we need it**:
The placeholder documentation introduces a couple of problems:
- it complicates the contributor-experience (forces the CI to run
N times before the contributor finds out that they need to call an .sh
script and include certain files from docs/)
- it forces CLI related pull requests for tools like kubeadm and kubectl
to require top level approval from docs/OWNERS as such PRs still need
to touch the .generated_docs file

Stop tracking the placeholder documentation by applying the
following actions:
- remove the utility set-placeholder-gen-docs()
- make verify-generated-docs.sh only generate in a temporary folder
and not match .generated_docs
- mark generate-docs.sh as an alias for update-generated-docs.sh
- remove all current placeholder files in docs folders admin, man,
user-guide, yaml
- ignore the above folders and .generated_docs in a .gitignore file

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubernetes#26205

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
^ ?

/assign @spiffxp @ixdy @lavalamp 
/priority important-longterm
/cc @kubernetes/sig-docs-maintainers 
/cc @kubernetes/sig-contributor-experience-pr-reviews 
/cc @timothysc @fabriziopandini @tengqm

### CODE REVIEW

File: hack/lib/util.sh [hunk 1]
```diff
@@ -221,21 +221,6 @@ kube::util::gen-docs() {
   popd > /dev/null
 }
 
-# Puts a placeholder for every generated doc. This makes the link checker work.
```
Comment (spiffxp) [inline hack/lib/util.sh]: so this looks like the reason we've been keeping these around? wonder where/what it is

**Review by spiffxp — COMMENTED** [review]
I'm all for getting rid of these, but I feel like we need some historical context from someone who knows why these are here.

As a consumer of the release tarball, I know these placeholders as a signal that I can run a script to get more docs. Is there some other doc that does this more effectively, or should we edit something else?

**Comment by neolit123** [comment]
> pull-kubernetes-verify — Job succeeded.

hmm....

File: hack/lib/util.sh (hunk 1 above)
Comment (lavalamp) [reply hack/lib/util.sh]: The reason we want the link checker to continue to work is that we want a guarantee that the links to continue to work.

I'm OK if you found another way to solve the problem, but this PR needs to explain that. See also: https://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence

Comment (neolit123) [reply hack/lib/util.sh]: i need to ask what is the "link checker" and what "links" are talking about?

the real docs are generated (and also HTML prettyfied) by sig-docs before a release and are published on the k8s website:
https://kubernetes.io/docs/reference/
what control does the k/k repo has over that?

i don't see the value of validating the existence of placeholders files listed in .generated_docs.
also, for the real files the list is populated based on file existence.

> Chesterton's fence is the principle that reforms should not be made until the reasoning behind the existing state of affairs is understood.

+1, let's understand all the reasons.

Comment (neolit123) [reply hack/lib/util.sh]: bump, i need some clarifications here.
thanks.

**Comment by neolit123** [comment]
just got a confirmation today that sig-docs and the website are no longer using `./hack/update-generated-docs.sh` or `./hack/generate-docs`:

> We were not generating the kubeadm docs via hack/generate-docs.sh script. The markdown file, thus the HTML, is difficult to read. We have been generating the docs using kubernetes-incubator/reference-docs.

https://github.com/kubernetes/website/pull/10960#issuecomment-440491123

which to my understanding leaves the output of these files to external, unknown consumers and further questions the existence of the placeholder generated docs.

cc @tengqm @Bradamant3

**Comment by tengqm** [comment]
> just got a confirmation today that sig-docs and the website are no longer using `./hack/update-generated-docs.sh` or `./hack/generate-docs`:
> 
> > We were not generating the kubeadm docs via hack/generate-docs.sh script. The markdown file, thus the HTML, is difficult to read. We have been generating the docs using kubernetes-incubator/reference-docs.
> 
> [kubernetes/website#10960 (comment)](https://github.com/kubernetes/website/pull/10960#issuecomment-440491123)
> 
> which to my understanding leaves the output of these files to external, unknown consumers and further questions the existence of the placeholder generated docs.
> 
> cc @tengqm @Bradamant3

Yes, that is true. We were using the output from these scripts before v1.9. The problem with the generated markdown files is that we only got a verbatim dump of the help message for the tools. In other words, it left us no room for improving the message for HTML layout. Starting from v1.10, we have been parsing the flags directly and generating HTML tables for tool options. Now the web pages are user friendly because the tables can be resized and scrolled.

We did considered the option to contribute this back to the upstream docs generator, however we are not sure how a HTML specific downstream generator fits into the whole picture.

**Comment by neolit123** [comment]
> We did considered the option to contribute this back to the upstream docs generator, however we are not sure how a HTML specific downstream generator fits into the whole picture.

ideally the generation of docs should happen at the same place. currently we have something that can be considered as a fork, even though it probably should be the default way.

something else that can be done is move the generation of reference docs outside of k/k and only leave it for validation. this also solves the problem in this PR.

then if someone wants to generate the official docs the can run a `./hack/....` script that pull this repository temporary generates the docs and removes it.

all of this can happen from a repo called `kubernetes/reference-docs` currently in incubator.

**Comment by spiffxp** [comment]
Based on the comments re: `kubernetes/reference-docs` and `hack/verify-linkcheck.sh` I surmise that no, none of these generated placeholders are actually being used.  Correct?

I think there are then two followup issues:
- do we bother attempting to revive the use of linkcheck? (I would vote either no, or yes but it's got to be pulled out of this repo into someplace like k/test-infra or its own repo in k-sigs/ so that it can also be used by k/community if not k/website)
- can we build a `kubernetes.tar.gz` that actually contains generated docs rather than the placeholders we have today?

WDYT?

File: hack/lib/util.sh (hunk 1 above)
Comment (neolit123) [reply hack/lib/util.sh]: +1

**Comment by neolit123** [comment]
> no, none of these generated placeholders are actually being used. Correct?

i believe so.

> do we bother attempting to revive the use of linkcheck? 

i don't think we have to. my question there would be - who are the consumers of the links that force the need for a link checker? if k/website is the only consumer they certainly don't need the placeholders, because they even have custom tooling to generate the docs.

> can we build a kubernetes.tar.gz that actually contains generated docs rather than the placeholders we have today?

the website contains the latest ref docs for each release.
given we have placeholders at master, people cannot reference our docs with links, so they have to download our docs and generate them manually.

we can potentially have the absolute bleeding edge, latest ref docs in a tgz, but i don't think this will have consumers and people just reference the website or generate manually.

**Comment by spiffxp** [comment]
/milestone v1.14

I would like to see these removed.  AFAIK all of the docs/OWNERS are unavailable at the moment.  Maybe next week.

In the meantime, needs rebase

**Comment by neolit123** [comment]
@spiffxp 
> In the meantime, needs rebase

i can easily rebase once this sees green light.

**Review by fabriziopandini — COMMENTED** [review]
I don't know the history behind doc placeholder, but for sure if it is possible to get rid of this step all PRs will gain in velocity (all the request are impacted by verify_generateddocs), and top level approvers will see less noise for the changes impacting cobra commands
LGTM

**Comment by timothysc** [comment]
@neolit123 bring this up on tomorrows agenda and lets close or try to get eyeballs on this.

**Comment by neolit123** [comment]
> @neolit123 bring this up on tomorrows agenda and lets close or try to get eyeballs on this.

sure.

**Comment by spiffxp** [comment]
@lavalamp is still out on leave AFAIK, I'm willing to take the heat on pushing this forward, it makes things no worse than before, and doesn't prevent us from trying to spin up a linkcheck job again

**Comment by pwittrock** [comment]
/approve

No longer tracking this stuff seems reasonable.

**Comment by spiffxp** [comment]
....

ok I'm going to fight the bot on this one, if it keeps putting the label back I'll look into a fresh blockade

**Comment by neolit123** [comment]
@spiffxp 
> ok I'm going to fight the bot on this one, if it keeps putting the label back I'll look into a fresh blockade

ok, this check seems to be new.
but the file `kubectl/kubectl.yaml` is still one of the auto-generated.

@pwittrock 
> No longer tracking this stuff seems reasonable.

thanks for the `/approve`.

**Comment by spiffxp** [comment]
The blockade this is hitting has been in place for a long time https://github.com/kubernetes/test-infra/pull/5302

You're running into it because you're now removing `docs/yaml/kubectl/kubectl.yaml` where you weren't before

Since we're nuking the generated docs, there's no need to blockade on changes to them.  I'll remove the blockade

**Comment by neolit123** [comment]
need to update again here. looks like i didn't remove all files.

**Comment by timothysc** [comment]
w00t!

**Comment by cjwagner** [comment]
I think this PR also needed to remove this line because it deletes the `docs/.generated_docs` file:
https://github.com/kubernetes/kubernetes/blob/903cdff633c16b8511a3bf1641f08ad0ee08e569/.generated_files#L29

Prow's `size` plugin is encountering errors because that file cannot be found: 
```yaml
{
  author:  "smarterclayton"   
  component:  "hook"   
  error:  "kubernetes/kubernetes/docs/.generated_docs @ b862590bfcd40d5f82f2a8e1d5d3d7b147b82d55 not found"   
  event-GUID:  "04853a00-2d7e-11e9-95d7-6790dd5027b3"   
  event-type:  "pull_request"   
  level:  "error"   
  msg:  "Error handling PullRequestEvent."   
  org:  "kubernetes"   
  plugin:  "size"   
  pr:  73892   
  repo:  "kubernetes"   
  url:  "https://github.com/kubernetes/kubernetes/pull/73892"   
 }
```

**Comment by neolit123** [comment]
@cjwagner 
i will send a PR for that in a bit and ping you.
not sure when it will get merged/approved as it's a top level file.

**Pull request merged**